gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -21,47 +21,73 @@
 import java.io.File;
 import java.util.AbstractList;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
-import net.ltgt.gradle.errorprone.ErrorPronePlugin;
+import net.ltgt.gradle.errorprone.javacplugin.CheckSeverity;
+import net.ltgt.gradle.errorprone.javacplugin.ErrorProneJavacPluginPlugin;
+import net.ltgt.gradle.errorprone.javacplugin.ErrorProneOptions;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.file.FileCollection;
+import org.gradle.api.plugins.ExtensionAware;
 import org.gradle.api.plugins.JavaPluginConvention;
 import org.gradle.api.tasks.compile.JavaCompile;
 import org.gradle.api.tasks.javadoc.Javadoc;
 import org.gradle.api.tasks.testing.Test;
 
 public final class BaselineErrorProne implements Plugin<Project> {
 
+    private static final String ERROR_PRONE_JAVAC_VERSION = "9+181-r4173-1";
+
     @Override
     public void apply(Project project) {
         project.getPluginManager().withPlugin("java", plugin -> {
-            project.getPluginManager().apply(ErrorPronePlugin.class);
-            project.getDependencies().add("errorprone", "com.palantir.baseline:baseline-error-prone:latest.release");
-            project.getTasks().withType(JavaCompile.class)
-                    .configureEach(compile -> compile.getOptions().getCompilerArgs()
-                            .addAll(ImmutableList.of(
-                                    "-XepDisableWarningsInGeneratedCode",
-                                    "-Xep:EqualsHashCode:ERROR",
-                                    "-Xep:EqualsIncompatibleType:ERROR")));
+            project.getPluginManager().apply(ErrorProneJavacPluginPlugin.class);
+            JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
+            String version = Optional.ofNullable(getClass().getPackage().getImplementationVersion())
+                    .orElse("latest.release");
+            project.getDependencies().add(
+                    ErrorProneJavacPluginPlugin.CONFIGURATION_NAME,
+                    "com.palantir.baseline:baseline-error-prone:" + version);
+
+
+            // This plugin interferes with gradles native annotation processor path configuration so need to configure
```

Comment:
any chance you could just fix the processors plugin?

Reply:
this is to avoid lockstep upgrading if you use that plugin and this. The real fix is to switch to gradles native annotationprocessor handling

Reply:
Created https://github.com/palantir/gradle-processors/pull/81 which should allow us to remove this code